### PR TITLE
LOOP-1492: Therapy Settings Delivery Limits: add max basal rate

### PR DESCRIPTION
**Pull request by rickpasetto**
Adds Max Basal Rate to the new Delivery Limits section (maximum bolus is coming next)
Note: this is dependent on PRs https://github.com/tidepool-org/LoopKit/pull/145 and https://github.com/tidepool-org/Loop/pull/158

[LOOP-1492](https://tidepool.atlassian.net/browse/LOOP-1492)

### CODE REVIEW

File: LoopKitUI/Views/Settings Editors/TherapySettingsView.swift [hunk 1]
```diff
@@ -195,18 +193,46 @@ extension TherapySettingsView {
...
-                if self.therapySettings.basalRateSchedule == nil {
-                    Text("NO basalRateSchedule")
+            }
+        }
+    }
+    
+    private var deliveryLimitsSection: some View {
+        section(for: .deliveryLimits) {
+            HStack {
+                Text(LocalizedString("Maximum Basal Rate", comment: "Maximum Basal Rate settings item title"))
```
Comment (mpangburn) [inline LoopKitUI/Views/Settings Editors/TherapySettingsView.swift]: Do we want the 'Maximum Basal Rate' text extracted?

File: LoopKitUI/Views/Settings Editors/TherapySettingsView.swift [hunk 2]
```diff
@@ -195,18 +193,46 @@ extension TherapySettingsView {
...
+                Spacer()
+                Group {
```
Comment (mpangburn) [inline LoopKitUI/Views/Settings Editors/TherapySettingsView.swift]: Use of Group here looks to be extraneous

**Review by mpangburn — APPROVED** [review]
Non-blocking question on text unification, otherwise LGTM

File: LoopKitUI/Views/Settings Editors/GuardrailExtensions.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
+//
+//  GuardrailExtensions.swift
```
Comment (novalegra) [inline LoopKitUI/Views/Settings Editors/GuardrailExtensions.swift]: [nit] I've noticed the name convention for extensions tends to be [thing you're extending]+[description]. Perhaps `Guardrail+Settings`?

File: LoopKitUI/Views/Settings Editors/TherapySettingsView.swift [hunk 4]
```diff
@@ -93,13 +87,13 @@ extension TherapySettingsView {
...
-                self.isEditing.toggle()
-            })
+            // TODO: confirm
+            self.delegate?.cancel()
+            self.viewModel.reset()
+            self.isEditing.toggle()
+        })
         {
-          Text(LocalizedString("Cancel", comment: "Cancel button text"))
+            Text(LocalizedString("Cancel", comment: "Cancel button text"))
```
Comment (novalegra) [inline LoopKitUI/Views/Settings Editors/TherapySettingsView.swift:96]: [nit] Spacing

File: LoopKitUI/Views/Settings Editors/TherapySettingsView.swift [hunk 5]
```diff
@@ -216,6 +242,7 @@ extension TherapySettingsView {
                         editAction: { self.delegate?.gotoEdit(therapySetting: therapySetting) },
                         content: content)
     }
+
```
Comment (novalegra) [inline LoopKitUI/Views/Settings Editors/TherapySettingsView.swift]: [nit] Extra space

**Review by novalegra — APPROVED** [review]
LGTM

File: LoopKitUI/Views/Settings Editors/GuardrailExtensions.swift (hunk 3 above)
Comment (rickpasetto) [reply LoopKitUI/Views/Settings Editors/GuardrailExtensions.swift]: @mpangburn had a similar comment for `TherapySettingExtensions`.  I'll change both to `+Settings` suffix.

**Pull request merged**